lib/puppet/resource/catalog.rb

```diff
@@ -398,6 +404,10 @@ def self.from_data_hash(data)
       result.catalog_uuid = catalog_uuid
     end
 
+    if catalog_format = data['catalog_format']
```

Comment:
If catalog_format isn't present, should we default to 0? For a newer agent talking to an older master.